File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxOperations = function(nums, k) {
    nums.sort((num1,num2)=>num1-num2)
    let i=0
    let j=nums.length-1
    let counter=0
    while(i<j){
        if(nums[i]>k){
            return counter
        }
        if(nums[i]+nums[j]===k){
            counter++
            i++
            j--
        }else if(nums[i]+nums[j]>k){
            j--
        }else{
            i++
        }
    }
    return counter
};

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        i=0
        j=len(nums)-1
        counter=0
        while i<j:
            if nums[i]>k:
                break
            if nums[i]+nums[j]==k:
                counter+=1
                i+=1
                j-=1
            elif nums[i]+nums[j]>k:
                j-=1
            else:
                i+=1
        return counter
        